packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -2808,7 +2826,9 @@ function updateDehydratedSuspenseComponent(
 
       // If we have scheduled higher pri work above, this will just abort the render
       // since we now have higher priority work. We'll try to infinitely suspend until
```

Comment:
This comment doesn't really make sense if this only happens for sync.

Reply:
Yeah the explanation is wrong. It's because if hydration suspends, we go back to the original update lane and switch to a client render. Then we suspend with delay.